### PR TITLE
Added order cleanup to e2e setup

### PR DESCRIPTION
**Pull request by zhongruige**
### Changes proposed in this Pull Request:

This PR introduces a cleanup orders function that permanently deletes all orders, regardless of status.

### How to test the changes in this Pull Request:

1. Verify the CI passes
2. Create a few simple orders in a variety of statuses (such as `pending`, `trash`, or `completed`). All possible order statuses are: `pending`, `processing`, `on-hold`, `completed`, `cancelled`, `refunded`, `failed`, `trash`.
3. Run the e2e tests with `npx wc-e2e test:e2e`
4. Open the `WooCommerce > Orders` page after the tests start running
5. Verify the orders you created get deleted

### CODE REVIEW

File: tests/e2e/utils/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 - Added `describeIf()` to conditionally run a test suite
 - Added `itIf()` to conditionally run a test case.
 - Added merchant workflows around plugins: `uploadAndActivatePlugin()`, `activatePlugin()`, `deactivatePlugin()`, `deletePlugin()`
+- Added `deleteAllOrders` that goes through and deletes all orders
```
Comment (rrennick) [inline tests/e2e/utils/CHANGELOG.md:15]: Add `()` to the function name for consistency.

File: tests/e2e/utils/src/flows/with-rest-api.js [hunk 2]
```diff
@@ -81,6 +82,24 @@ export const withRestApi = {
 		const repository = SimpleProduct.restRepository( client );
 		await deleteAllRepositoryObjects( repository );
 	},
+	/**
+	 * Use api package to delete all orders.
+	 *
+	 * @return {Promise} Promise resolving once orders have been deleted.
+	 */
+	deleteAllOrders: async () => {
```
Comment (rrennick) [inline tests/e2e/utils/src/flows/with-rest-api.js:90]: `deleteAllRepositoryObjects()` could be updated to accept a status array as a parameter. That would eliminate duplicating the logic.

**Review by rrennick — CHANGES_REQUESTED** [review]
@zhongruige This worked great locally. I monitored the order list while running the full test suite and it deleted the existing orders before each test. This should be good to merge with the suggestions I've added.

File: tests/e2e/utils/CHANGELOG.md (hunk 1 above)
Comment (zhongruige) [reply tests/e2e/utils/CHANGELOG.md]: Nice catch, fixed!

File: tests/e2e/utils/src/flows/with-rest-api.js (hunk 2 above)
Comment (zhongruige) [reply tests/e2e/utils/src/flows/with-rest-api.js:91]: I prefer that. I updated the delete to use this:

```javascript
deleteAllOrders: async () => {
	// We need to specfically filter on order status here to make sure we catch all orders to delete.
	const orderStatuses = ['pending', 'processing', 'on-hold', 'completed', 'cancelled', 'refunded', 'failed', 'trash'];
	const repository = Order.restRepository( client );
	await deleteAllRepositoryObjects( repository, null, orderStatuses );
},
``

And have the following set of default statuses being passed in:

```javascript
const deleteAllRepositoryObjects = async ( repository, defaultObjectId = null, statuses = [ 'draft', 'publish', 'trash' ] ) => {}
```

**Review by rrennick — DISMISSED** [review]
@zhongruige Nice work. This tested great locally.

I have pre-approved. Feel free to merge once the errant leading space is fixed.

**Comment by rrennick** [comment]
I restarted CI on this as a couple of the checked failed.

File: tests/e2e/utils/src/flows/with-rest-api.js [hunk 3]
```diff
@@ -12,12 +12,12 @@ const userEndpoint = '/wp/v2/users';
  *
  * @param repository
  * @param defaultObjectId
+ * @param statuses Status of the object to check
  * @returns {Promise<void>}
  */
-const deleteAllRepositoryObjects = async ( repository, defaultObjectId = null ) => {
+ const deleteAllRepositoryObjects = async ( repository, defaultObjectId = null, statuses = [ 'draft', 'publish', 'trash' ] ) => {
```
[the remark this answers is not in the document]
Comment (zhongruige) [reply tests/e2e/utils/src/flows/with-rest-api.js]: Nice catch, removed!

**Comment by rrennick** [comment]
@zhongruige I'll merge this. The check that's failing isn't related to this PR.

**Pull request merged**